packages/fxa-auth-server/lib/routes/subscriptions/utils.ts

```diff
@@ -45,10 +45,8 @@ export function handleUidAuth(auth: AuthRequest['auth']): string {
 
 export function handleAuthScoped(auth: AuthRequest['auth'], scopes: string[]) {
   const scope = ScopeSet.fromArray(auth.credentials.scope);
-  for (const requiredScope of scopes) {
-    if (!scope.contains(requiredScope)) {
-      throw error.invalidScopes();
-    }
+  if (!scopes.some((requiredScope) => scope.contains(requiredScope))) {
+    throw error.invalidScopes();
```

Comment:
`handleAuthScoped` previously had a bug where if the request scope didn't contain the first required scope in the array, it would error. I’ve since fixed this to where the request scope has to contain any of the required scopes in the list.

However, there was no test for this helper (nor for `handleUidAuth` also in this file), and I noticed that `handleAuth`'s test is in `test/local/routes/subscriptions/stripe.js` despite being used by a number of other routes, so I plan to file a follow up ticket to write tests for these helpers and move all these tests to a `test/local/routes/subscriptions/utils.js` file or similar, as that is outside the scope of this ticket (and we're running against the clock as it is).

Reply:
I'm not sure this is a bug, I think we intentionally have it set so that you must have all the scopes in requiredScopes. For routes that are more variable, like our profile route that includes varying data depending on which scopes you have, it does scope tests before including each bit of info, and only has the minimum scope required set on the route itself.

Reply:
@vbudhram Is my understanding there correct, or is this check not working as intended?

Reply:
For the application here, the VPN client will have either the `"profile"` scope (in the case of migrating existing Apple IAP users) or the `OAUTH_SCOPE_SUBSCRIPTIONS_IAP` scope (in the case of a logged in user creating an IAP subscription).

Reply:
@biancadanforth I think we could probably have some manual checking that this user is in either scope and comments that we can remove once migration is completed.